downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+- Extension of Ballerina transaction capabilities to define transactional resource functions and transactional remote functions
+- Improvements to the HTTP, Log, Email, WebSub, and UUID Standard Library modules
+- Introduction of the new GraphQL, NATS Streaming (STAN), and WebSocket Standard Library modules
+
+### What is new in Ballerina Swan Lake Preview 8
+
+#### Language
+
+##### Support Identifier Escapes Without an Initial Quote
+
+Now, `\` can simply be used to escape the special characters in an identifier without mentioning the initial quote. For example, both the formats below are supported now.
+
+```ballerina
+int a\-b;
+int 'a\-b;
+```
+
+##### Included Record Parameters
+
+Included record parameters can be specified as `*T P` in which `T` denotes the record type descriptor and `P` denotes the name of the parameter. When this is compared with the required parameter, the difference is that the values, which are used for the fields of the included record parameter behave as named arguments for the function call, if it had been declared as parameters of the function.
+
+The field names of all included record parameters, which are not of the `never` type, must be distinct from each other and also from the names of the parameters.
+
+```ballerina
+type Student record {
+	string firstName;
+	string secondName;
+	int age;
+	Address address?;
+	never gender?;
+};
+
+type Address record {|
+	string city;
+	string country;
+|};
+
+type Grades record {|
+	never maths?;
+	never physics?;
+	never chemistry?;
+
+	int...;
+|};
+
+function studentDetails(int addmissionNo, string addmissionDate, *Student student) {
+	string fullName = <string>student.firstName  + <string>student.secondName;
+	int age = student.age;
+	if (student["address"] is Address) {
+    	Address address = <Address>student["address"];
+	}
+}   
+
+function studentTotalMarks(int maths, int physics, int chemistry, *Grades grades) {
+	int totalMarks = maths + physics + chemistry;
+	foreach [string, int?] [subject, marks] in grades.entries() {
+    	totalMarks = totalMarks + <int>marks;
+	}
+}
+```
+
+A named argument in a function call can correspond to an included record parameter in two different ways.
+
+1. Being relevant to a field of the included recorded parameter, which is not of the `never` type.
+
+    ```ballerina 
+    public function main() {
+        studentDetails(1001, "2020-Nov-20", firstName = "Peter", secondName = "So", age = 18);
+        studentDetails(1002, "2020-Nov-22", firstName = "Anne", secondName = "Doe", age = 20, address = {city: "Colombo", country: "Sri Lanka"});
+    }
+    ```
+
+2. Being relevant to a record rest descriptor of an included record parameter. Here, the field is described by a record rest descriptor either as explicitly part of an exclusive record type descriptor or as implicitly as part of an inclusive record type descriptor. The two conditions below should be satisfied for this.
```

Comment:
@chiranSachintha can't we demonstrate these with just one function?

```ballerina
type Grades record {|
    never math?;
    int...;
|};

function getTotalMarks(int math, *Grades grades) returns int {
    int totalMarks = math;

    foreach int marks in grades {
        totalMarks += marks;
    }

    return totalMarks;
}
```

Reply:
I have created two functions for the two scenarios. 

1. Belonging named argument to a field of an included record parameter
2. Belonging named argument to a field of an included record parameter as an additional field(allowed by the record being open).
Isn’t it better to use two functions than using one function?

Reply:
We can still cover both scenarios with something like

```
type Grades record {|
    never math?;
    int physics;
    int...;
|};

function getTotalMarks(int math, *Grades grades) returns int {
    int totalMarks = math + grades.physics;

    foreach int marks in grades {
        totalMarks += marks;
    }

    return totalMarks;
}
```

right?

IMO, we should keep the example in the release note as concise as possible so that anyone going through it can quickly grasp the changes. A BBE on the other hand can go into more detail with multiple functions.

Reply:
Changes in https://github.com/praneesha/ballerina-dev-website/pull/10 will address these.